bin/add-cmderplugin.ps1

```diff
@@ -0,0 +1,161 @@
+
+    Then unblock the script for execution with UnblockFile .\add-cmderplugin.ps1
+.EXAMPLE
+    .\add-cmderplugin.ps1 -PluginPath '~/custom/cmder-plugin-[name].json'
+
+    Reads the source json data and excutes the steps to install nad configue the plugin.
+.EXAMPLE
+    .\add-cmderplugin.ps1 -PluginPath '~/custom/cmder-plugin-[name].json' -verbose
+
+    Reads the source json data and excutes the steps to install nad configue the plugin and see more detail as to what is going on.
+.NOTES
+    AUTHORS
+    Dax Games, Samuel Vasko, Jack Bennett
+    Part of the Cmder project.
+.LINK
+    http://cmder.net/ - Project Home
+#>
+
+[CmdletBinding(SupportsShouldProcess=$true)]
+Param(
+    # CmdletBinding will give us;
+    # -verbose switch to turn on logging and
+    # -whatif switch to not actually make changes
+
+    # Path to the vendor configuration source file
+    [string]$PluginPath = "$env:CMDER_ROOT\vendor\sources-plugins.json",
+
+    # Replace the plugin
+    [switch]$force,
+
+    # Vendor folder location
+    [string]$saveTo = "$env:CMDER_ROOT\vendor\",
+
+    # Config folder location
+    [string]$config = "$env:CMDER_ROOT\config"
+)
+
+
+$ThisScript = Get-Item $MyInvocation.MyCommand.Definition
+
+# Dot source util functions into this scope
+. $($ThisScript.DirectoryName + "\utils.ps1")
+
+$bash_user_profile = $($env:cmder_root + "\config\user-profile.sh")
+$bash_user_aliases = $($env:cmder_root + "\config\user-aliases.sh")
+
+if (test-path $bash_user_aliases) {
+  $bash_aliases = $bash_user_aliases
+} else {
+  $bash_aliases = $bash_user_profile
+}
+
+$posh_user_profile = $($env:cmder_root + "\config\user-profile.ps1")
+$posh_user_aliases = $($env:cmder_root + "\config\user-aliases.ps1")
+
+if (test-path $posh_user_aliases) {
+  $posh_aliases = $posh_user_aliases
+} else {
+  $posh_aliases = $posh_user_profile
+}
+
+
+$cmder_user_profile = $($env:cmder_root + "\config\user-profile.cmd")
+$cmder_aliases = $($env:cmder_root + "\config\aliases")
+
+$ErrorActionPreference = "Stop"
+
+Push-Location -Path $saveTo
+$sources = Get-Content $PluginPath | Out-String | Convertfrom-Json
+
+# Check for requirements
+Ensure-Exists $PluginPath
+Ensure-Executable "7z"
+New-Item -Type Directory -Path (Join-Path $saveTo "/tmp/") -ErrorAction SilentlyContinue >$null
+
+$vend = $pwd
+foreach ($s in $sources) {
+    Write-Verbose "Getting $($s.name) from URL $($s.url)"
+
+    foreach ($pre_command in $s.pre_commands) {
+      write-host $("`nRunning '" + $pre_command + "'...")
+      invoke-expression $pre_command
+    }
+
+    # We do not care about the extensions/type of archive
+    $tempArchive = "tmp/$($s.name).tmp"
+    Delete-Existing $tempArchive
+
+    if ($force -eq $true) {
+      write-host $("Deleting existing install of the '" + $s.name + "' plugin...")
+      Delete-Existing $s.name
+    }
+
+    if ((test-path $s.name) -eq $false) {
```

Comment:
Shouldn't this also do some test for the version and only complain if the same (or a higher) version number is already installed?